javascript/ql/src/semmle/javascript/Regexp.qll

```diff
@@ -160,68 +223,193 @@ class RegExpSequence extends RegExpTerm, @regexp_seq {
+ */
 class RegExpNonWordBoundary extends RegExpTerm, @regexp_nonwordboundary {
   override predicate isNullable() { any() }
 }
 
-/** A zero-width lookahead or lookbehind assertion. */
+/**
+ * A zero-width lookahead or lookbehind assertion.
+ *
+ * Examples:
+ *
+ * ```
+ * (?=\w)
+ * (?!\n)
+ * (?<=\.)
+ * (?<!\\)
+ * ```
+ */
 class RegExpSubPattern extends RegExpTerm, @regexp_subpattern {
   /** Gets the lookahead term. */
   RegExpTerm getOperand() { result = getAChild() }
 
   override predicate isNullable() { any() }
 }
 
-/** A zero-width lookahead assertion. */
+/**
+ * A zero-width lookahead assertion.
+ *
+ * Examples:
+ *
+ * ```
+ * (?=\w)
+ * (?!\n)
+ * ```
+ */
 class RegExpLookahead extends RegExpSubPattern, @regexp_lookahead { }
 
-/** A zero-width lookbehind assertion. */
+/**
+ * A zero-width lookbehind assertion.
+ *
+ * Examples:
+ *
+ * ```
+ * (?<=\.)
+ * (?<!\\)
+ * ```
+ */
 class RegExpLookbehind extends RegExpSubPattern, @regexp_lookbehind { }
 
-/** A positive-lookahead assertion, that is, a term of the form `(?=...)`. */
+/**
+ * A positive-lookahead assertion.
+ *
+ * Examples:
+ *
+ * ```
+ * (?=\w)
+ * ```
+ */
 class RegExpPositiveLookahead extends RegExpLookahead, @regexp_positive_lookahead { }
 
-/** A negative-lookahead assertion, that is, a term of the form `(?!...)`. */
+/**
+ * A negative-lookahead assertion.
+ *
+ * Examples:
+ *
+ * ```
+ * (?!\n)
+ * ```
+ */
 class RegExpNegativeLookahead extends RegExpLookahead, @regexp_negative_lookahead { }
 
-/** A positive-lookbehind assertion, that is, a term of the form `(?<=...)`. */
+/**
+ * A positive-lookbehind assertion.
+ *
+ * Examples:
+ *
+ * ```
+ * (?<=\.)
+ * ```
+ */
 class RegExpPositiveLookbehind extends RegExpLookbehind, @regexp_positive_lookbehind { }
 
-/** A negative-lookbehind assertion, that is, a term of the form `(?<!...)`. */
+/**
+ * A negative-lookbehind assertion.
+ *
+ * Examples:
+ *
+ * ```
+ * (?<!\\)
+ * ```
+ */
 class RegExpNegativeLookbehind extends RegExpLookbehind, @regexp_negative_lookbehind { }
 
-/** A star-quantified term, that is, a term of the form `...*`. */
+/**
+ * A star-quantified term.
+ *
+ * Example:
+ *
+ * ```
+ * \w*
+ * ```
+ */
 class RegExpStar extends RegExpQuantifier, @regexp_star {
   override predicate isNullable() { any() }
 }
 
-/** A plus-quantified term, that is, a term of the form `...+`. */
+/**
+ * A plus-quantified term.
+ *
+ * Example:
+ *
+ * ```
+ * \w+
+ * ```
+ */
 class RegExpPlus extends RegExpQuantifier, @regexp_plus {
   override predicate isNullable() { getAChild().isNullable() }
 }
 
-/** An optional term, that is, a term of the form `...?`. */
+/**
+ * An optional term.
+ *
+ * Example:
+ *
+ * ```
+ * ;?
+ * ```
+ */
 class RegExpOpt extends RegExpQuantifier, @regexp_opt {
   override predicate isNullable() { any() }
 }
 
-/** A range-quantified term, that is, a term of the form `...{m,n}`. */
+/**
+ * A range-quantified term
+ *
+ * Example:
+ *
+ * ```
+ * [0-9a-fA-F]{2,4}
```

Comment:
Unfortunate example, given that `a-f` and `A-F` are also "ranges". Maybe just pick a simpler base expression.